lib/active_admin/namespace.rb

```diff
@@ -37,14 +37,23 @@ def setting(name, default)
 
     def initialize(application, name)
       @application = application
-      @name = name.to_s.underscore
+      @name = if name.is_a?(Array)
+                default_namespace = application.default_namespace
+                [:root, false, nil].include?(default_namespace) || (name.first == default_namespace) ? name : [default_namespace] + name
+              else
+                name.to_s.underscore
+              end
       @resources = ResourceCollection.new
       register_module unless root?
       build_menu_collection
     end
 
     def name
-      @name.to_sym
+      if @name.is_a?(Array)
+        @name
+      else
+        @name.to_sym
+      end
     end
```

Comment:
I propose treating this as a breaking change as follows:
```ruby
    def name
      Deprecation.warn "name replaced by namepath now that namespaces can be nested."
      namepath.first
    end

    # @param namepath [Array]
    def namepath=(namepath)
      namepath.shift if namepath.first == :root
      @namepath = namepath.map(&:to_s).map(&:underscore)
      default_namespace = application.default_namespace
      if default_namespace && default_namespace != :root
        @namepath.unshift default_namespace.to_s unless @namepath.first == default_namespace.to_s
      end
    end

    def namepath
      @namepath.map(&:to_sym)
    end
```
This approach leaves #name always returning the same symbol as before but warning of new functionality.  As #name is replaced by #namepath the latter always returns an array of symbols so no conditional logic is required to test the type of value returned.

Reply:
Does that mean we will assign the namepath in initialize? And why do we need to convert the namepath into a underscored string?

Reply:
Yes, call namepath= from initialize.

Reply:
Sure, I get the idea. And do you mind having a look of my recent changes? I can add the breaking changes on top.